lib/super_parser.rb

```diff
@@ -0,0 +1,64 @@
+require 'webdrivers'
+require 'watir'
+require 'nokogiri'
+require 'pry'
+require 'json'
+
+require_relative 'super_parser/accounts'
+require_relative 'super_parser/transactions'
+
+class SuperParser
+  BASE_URL = 'https://my.fibank.bg/'.freeze
+
+  attr_accessor :browser, :account_details, :last_transactions
+
+  def initialize
+    web_connect
+    visit_profile_account_page
+    get_accounts_details
+    get_last_transactions
+    browser.close
+
+    puts account_details
+    puts last_transactions
+  end
+
+
+  def web_connect
+    @browser = Watir::Browser.new
+    @browser.goto BASE_URL
+  end
+
+  def visit_profile_account_page
+    browser.link(id: 'demo-link').click
```

Comment:
What happens if after this click the page didn't load? Or there is a spinner? You should wait for a specific element to appear on page. In this case, you should wait the element with accounts to load.

Reply:
Take into consideration this comment for all places in your code when you navigate to a page. You need to ensure that the page changed and it is indeed the page you expected.